include/gridtools/common/int_vector.hpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * Implements operations on `int_vector`s, which are hymaps of integers (integral types or integral_constants)
+ *
+ * A type `T` is `int_vector` if
+ *  - `is_hymap<T>`
+ *  - all value types `V` are std::is_integral<V> or is_gr_integral_constant<V>
+ *  - `std::is_trivially_copy_constructible<T>`
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include "../meta.hpp"
+#include "defs.hpp"
+#include "host_device.hpp"
+#include "hymap.hpp"
+#include "integral_constant.hpp"
+#include "tuple_util.hpp"
+#include "utility.hpp"
+
+namespace gridtools {
+
+    namespace int_vector::impl_ {
+        template <class... Ts>
+        struct value_t_merger;
+
+        template <template <class...> class Item, class Key, class... Vs>
+        struct value_t_merger<Item<Key, Vs>...> {
+            using type = Item<Key, std::decay_t<decltype((Vs{} + ...))>>;
+        };
+
+        template <class... Ts>
+        using merger_t = typename value_t_merger<Ts...>::type;
+
+        template <class Key>
+        struct add_f {
+            template <class... Ts>
+            GT_FUNCTION GT_CONSTEXPR decltype(auto) operator()(Ts const &...args) const {
+                return (host_device::at_key_with_default<Key, integral_constant<int, 0>>(args) + ...);
+            }
+        };
+
+        template <class I>
+        using is_constant_zero = is_integral_constant_of<meta::second<I>, 0>;
+
+        template <class Key>
+        struct at_key_f {
+            template <class T>
+            GT_FUNCTION GT_CONSTEXPR decltype(auto) operator()(T const &arg) const {
+                return host_device::at_key<Key>(arg);
+            }
+        };
+
+        template <class T>
+        using is_integral_or_gr_integral_constant =
+            bool_constant<std::is_integral<T>::value || is_gr_integral_constant<T>::value>;
+
+        template <class T>
+        using elements_are_integral_or_gr_integral_constant =
+            meta::all<meta::transform<int_vector::impl_::is_integral_or_gr_integral_constant,
+                meta::transform<meta::second, hymap::to_meta_map<T>>>>;
+
+    } // namespace int_vector::impl_
+
+    template <class T, class = void>
+    struct is_int_vector : std::false_type {};
```

Comment:
I would prefer to have `is_int_vector` within `gridtools::int_vector` namespace and export it to the outer `gridtools` via `using`. The same is for `is_int_vector_v`.